### PR TITLE
Go back to one data bag item per repository

### PR DESCRIPTION
**Pull request by vuntz**
cc @tboerger @MaximilianMeister

### CODE REVIEW

**Comment by aspiers** [comment]
The first commit is missing a justification for the change.  In particular I'm curious to understand why having multiple data bags prevents the race conditions @tboerger mentioned.

Also I'm wondering how the second commit is related to this PR?

**Comment by vuntz** [comment]
> Also I'm wondering how the second commit is related to this PR?

It's not, I moved it to https://github.com/crowbar/crowbar-core/pull/114

**Comment by vuntz** [comment]
> The first commit is missing a justification for the change.

I improved the commit message for this.

File: crowbar_framework/app/models/provisioner_service.rb [hunk 1]
```diff
@@ -279,16 +267,25 @@ def disable_repository(platform, arch, repo, bag = nil)
...
       @logger.debug("#{repo_id} repository for #{platform} / #{arch} is already inactive.")
     end
 
     [200, ""]
   end
+
+  private
+
+  # workaround for Chef::DataBag.destroy not working
+  def chef_data_bag_destroy(name)
```
Comment (tboerger) [inline crowbar_framework/app/models/provisioner_service.rb]: Why do you need that? My tests showed that destroying a data bag works fine.

Comment (vuntz) [inline crowbar_framework/app/models/provisioner_service.rb]: Interesting, that doesn't work for me. Can you try the patch and use the proper destroy method to see if that works for you?

Comment (tboerger) [inline crowbar_framework/app/models/provisioner_service.rb]: You are right... Maybe i mixed it up with data_bag items. For reference: https://github.com/chef/chef/blob/10-stable/chef/lib/chef/knife/data_bag_delete.rb#L39

**Comment by vuntz** [comment]
@tboerger reminder: you wanted this, so maybe a +1/-1 would be good ;-)

**Comment by tboerger** [comment]
Can not fine real mistakes. :+1:

**Comment by rsalevsky** [comment]
:+1:

**Pull request merged**